src/coprocessor/codec/data_type/mod.rs

```diff
@@ -23,3 +27,100 @@ pub use crate::coprocessor::codec::mysql::{Decimal, Duration, Json, Time as Date
 // Dynamic eval types.
 pub use self::scalar::ScalarValue;
 pub use self::vector::VectorValue;
+pub use self::vector_like::{VectorLikeValueRef, VectorLikeValueRefSpecialized};
+
+/// A trait of evaluating current concrete eval type into a MySQL logic value, represented by
+/// Rust's `bool` type.
+pub trait AsMySQLBool {
+    /// Evaluates into a MySQL logic value.
+    fn as_mysql_bool(&self) -> bool;
```

Comment:
I think we should return ` Result<bool>`

Reply:
Shall we only return warnings? Looks wired if we return error here (because it means that the selection will fail?)